In [2]:
import numpy as np
import pandas as pd

%matplotlib inline
import matplotlib.pyplot as plt
plt.style.use('fivethirtyeight')

<center> 

# Introduction to Statistics

## Amanda R. Kube Jotte
    
<img src="https://raw.githubusercontent.com/amandakube/Data118LectureImages/main/DSSI.png" width="800">
    
</center>

## Remember those penguins?

<center>
<img src="https://pbs.twimg.com/media/EaAWkZ0U4AA1CQf?format=jpg&name=4096x4096" width="900">
</center>

In [11]:
penguins = pd.read_csv("https://raw.githubusercontent.com/amandakube/Data118LectureImages/main/penguins_size.csv")
penguins.head(10)

species     island  culmen_length_mm  culmen_depth_mm  flipper_length_mm   
0  Adelie  Torgersen              39.1             18.7              181.0  \
1  Adelie  Torgersen              39.5             17.4              186.0   
2  Adelie  Torgersen              40.3             18.0              195.0   
3  Adelie  Torgersen               NaN              NaN                NaN   
4  Adelie  Torgersen              36.7             19.3              193.0   
5  Adelie  Torgersen              39.3             20.6              190.0   
6  Adelie  Torgersen              38.9             17.8              181.0   
7  Adelie  Torgersen              39.2             19.6              195.0   
8  Adelie  Torgersen              34.1             18.1              193.0   
9  Adelie  Torgersen              42.0             20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       3625.0  FEMALE  
7       4675.0    MALE  
8       3475.0     NaN  
9       4250.0     NaN

In [13]:
penguins.shape

(344, 7)

## I just learned what a culmen is and now I'm interested

What is the average culmen length in millimeters for a penguin in this dataset?

In [14]:
penguins.culmen_length_mm.mean()

43.9219298245614

## It's cold in Antarctica!

I dont want to stay long enough to collect data on all 344 penguins...

Can I learn something about culmen length from fewer penguins?

Imagine our dataset contains all the penguin information that I could gather on those three islands.

I go to the islands and stay until I've measured 100 penguins.

Pandas has a method `.sample()` that randomly selects rows from a dataframe.

In [15]:
# Imagine this is the data I gather
my_sample = penguins.sample(100)
my_sample.head()

species  island  culmen_length_mm  culmen_depth_mm  flipper_length_mm   
85      Adelie   Dream              41.3             20.3              194.0  \
274     Gentoo  Biscoe              46.5             14.4              217.0   
159  Chinstrap   Dream              51.3             18.2              197.0   
326     Gentoo  Biscoe              41.7             14.7              210.0   
29      Adelie  Biscoe              40.5             18.9              180.0   

     body_mass_g     sex  
85        3550.0    MALE  
274       4900.0  FEMALE  
159       3750.0    MALE  
326       4700.0  FEMALE  
29        3950.0    MALE

In [16]:
my_sample.culmen_length_mm.mean() # Is this a good estimate?

43.82653061224491

## Let's explore further...

Split into 6 groups

Repeat this experiment 1000 times (use a for loop!) and create a histogram of your results. When finished, save your histogram and send to `akube@uchicago.edu` with your group number in the subject so I can share with the class.

Group 1: Use the `.sample()` method to randomly choose 100 penguins

Group 2: Use the `.sample()` method to randomly choose 50 penguins

Group 3: Use the `.sample()` method to randomly choose 10 penguins

Group 4: Use the `.sample()` method to randomly choose 50 penguins from **only Adelie penguins**

Group 5: Use the `.sample()` method to randomly choose 50 penguins from **only male penguins**

Group 6: Use the `.sample()` method to randomly choose 50 penguins from **only penguins living on Dream**

In [ ]:
# sample code for your experiment, fill in the missing pieces

outcomes = ...

for ...:
    ...
    
plt.hist(...)
plt.title("...")

#do not change code below!
plt.xlim([38, 50])
plt.scatter(penguins.culmen_length_mm.mean(), -5, color='red', s=30)
plt.xlabel("Average Culmen Length (mm)")
plt.ylabel("Frequency");

Let's look at your results:

Group 1: Use the `.sample()` method to randomly choose 100 penguins

<center>
<img src="" width="700">
</center>

Group 2: Use the `.sample()` method to randomly choose 50 penguins

<center>
<img src="" width="700">
</center>

Group 3: Use the `.sample()` method to randomly choose 10 penguins

<center>
<img src="" width="700">
</center>

Group 4: Use the `.sample()` method to randomly choose 50 penguins from **only Adelie penguins**

<center>
<img src="" width="700">
</center>

Group 5: Use the `.sample()` method to randomly choose 50 penguins from **only male penguins**

<center>
<img src="" width="700">
</center>

Group 6: Use the `.sample()` method to randomly choose 50 penguins from **only penguins living on Dream**

<center>
<img src="" width="700">
</center>

## Which samples were more consistent with our true average culmen length? Why?

## What you just did is the basis for the whole concept of statistical inference!

We have some **population** we want to learn about - all penguins

We can't (infeasible, expensive, impossible) measure the whole population - **census**

We can measure a smaller portion, or **sample**, of that population

What we want to calculate/know about a **p**opulation is called the **parameter**

We estimate the parameter by calculating this on the **s**ample instead - a **statistic**

What was our statistic in the activity?

What was our parameter?

Which experiment gave the best estimates of our parameter? Why?

We will dig further into this in the morning and talk about what makes a **representative** sample!